src/stream/src/executor/source/fs_source_executor.rs

```diff
@@ -386,11 +393,7 @@ impl<S: StateStore> FsSourceExecutor<S> {
                     chunk,
                     split_offset_mapping,
                 }) => {
-                    if last_barrier_time.elapsed().as_millis() > max_wait_barrier_time_ms {
-                        // Exceeds the max wait barrier time, the source will be paused. Currently
-                        // we can guarantee the source is not paused since it received stream
-                        // chunks.
-                        self_paused = true;
+                    if !stream.paused() && self.throttlers.iter().any(|t| t.should_pause()) {
                         stream.pause_source();
```

Comment:
Please add `warn!` logging to inform us/users that streaming is paused due to accumulated L0 SST files.